flink-core/src/test/java/org/apache/flink/configuration/TaskManagerLoadBalanceModeTest.java

```diff
@@ -1,49 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.apache.flink.configuration;
-
-import org.junit.jupiter.api.Test;
-
-import static org.apache.flink.configuration.TaskManagerOptions.TASK_MANAGER_LOAD_BALANCE_MODE;
-import static org.apache.flink.configuration.TaskManagerOptions.TaskManagerLoadBalanceMode;
-import static org.assertj.core.api.Assertions.assertThat;
-
-/** Test for {@link TaskManagerLoadBalanceMode}. */
-class TaskManagerLoadBalanceModeTest {
-
-    @Test
-    void testReadTaskManagerLoadBalanceMode() {
```

Comment:
I do not think we should remove tests of deprecated methods - which are still supported. I think we should remove the tests at the same time as the methods / constants are removed

Reply:
thanks @davidradl  for the comments.
Please let me clarify the background fo the testing class.

The test was initially introduced to verify whether the new configuration method is compatible with the old one. Since the old configuration method has now been removed, the logic for determining the value of the current configuration item has become much simpler and more straightforward. In other words, continuing to keep this test case would be somewhat redundant.

WDYTA?

Reply:
@RocMarshal So the test is testing values that cannot longer be set? But the enumeration values still exist. Can TaskManagerLoadBalanceMode.SLOTS or TaskManagerOptions.TASK_MANAGER_LOAD_BALANCE_MODE be removed?

Reply:
Thanks @davidradl 
> TaskManagerOptions.TASK_MANAGER_LOAD_BALANCE_MODE be removed?
It is evident that these enumeration values cannot be removed, but this does not necessarily mean that this test case cannot be deleted.

Let me put it another way: the logic for setting and retrieving values in the current test case actually falls under the responsibility of the configuration class. In the corresponding test class for the configuration class, I believe sufficient path coverage tests have already been conducted to ensure that the configuration class can correctly set and retrieve values, including handling configuration values of enumeration types.

The enumeration values in the current test case are merely specific examples, and I don't think they differ fundamentally from the test cases in the configuration test class. Furthermore, if adding a new enumeration-type configuration value always requires creating a test case like the current one, it could lead to over-testing and unnecessarily increase the number of related test classes.

Pls let me know what's your opinion~

Reply:
Ok that makes sense